cmake/setup_finalize.cmake

```diff
@@ -84,12 +84,15 @@ ENDFOREACH()
 #
 
 FOREACH(build ${DEAL_II_BUILD_TYPES})
-  CHECK_COMPILER_SETUP(
-    "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${build}}"
-    "${DEAL_II_LINKER_FLAGS} ${DEAL_II_LINKER_FLAGS_${build}}"
-    DEAL_II_HAVE_USABLE_FLAGS_${build}
-    ${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${build}}
-    )
+# FIXME: until https://github.com/dealii/dealii/issues/3686 is resolved,
+# the simple tests below hangs and renders deal.II unusable.
+  SET(DEAL_II_HAVE_USABLE_FLAGS_${build} TRUE)
+#  CHECK_COMPILER_SETUP(
+#    "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${build}}"
+#    "${DEAL_II_LINKER_FLAGS} ${DEAL_II_LINKER_FLAGS_${build}}"
+#    DEAL_II_HAVE_USABLE_FLAGS_${build}
+#    ${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${build}}
+#    )
```

Comment:
What about, you simply remove the comment and readd the test, but remove the link interface, i.e.
`${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${build}}` from the invocation of `CHECK_COMPILER_SETUP`?

This is a very important check that shouldn't be disabled lightly - ask @bangerth, or @tjhei how many frustrated users we had in the past that figured out after 2h of compilation that they couldn't link because of some stupid mistake in the setup. At minimum we have to check for a usable compiler and linker setup.

Please check whether this change works in your spack eco system (it's good that someone can reproduce that.)